x-pack/test/functional/services/ml/job_table.ts

```diff
@@ -493,6 +516,14 @@ export function MachineLearningJobTableProvider(
       await testSubjects.existOrFail('mlJobEditFlyout');
     }
 
+    public async clickEditAnnotationAction(jobId: string) {
+      await testSubjects.click(this.detailsSelector(jobId, 'euiCollapsedItemActionsButton'));
```

Comment:
As mentioned in the last review: we have a helper method for this already: `ensureJobActionsMenuOpen`

Reply:
Roger that.

Reply:
Actually, I'm a bit confused boss.
How does this help me?

```
    public async ensureJobActionsMenuOpen(jobId: string) {
      await retry.tryForTime(30 * 1000, async () => {
        if (!(await testSubjects.exists('mlActionButtonDeleteJob'))) {
          await testSubjects.click(this.rowSelector(jobId, 'euiCollapsedItemActionsButton'));
          await testSubjects.existOrFail('mlActionButtonDeleteJob', { timeout: 5000 });
        }
      });
    }
```
This raises an eyebrow for me: `if (!(await testSubjects.exists('mlActionButtonDeleteJob'))) {`

Reply:
Oh, perhaps that's "backstop".

Reply:
Hrmm, so my version uses `this.detailsSelector` while ensureJobActionsMenuOpen() uses `this.rowSelector`.

This does not work boss, fails with:

```
1 failing

1)    machine learning - anomaly detection
       job list
         expanded row with annotations can be edited:

      Error: retry.try reached timeout 120000 ms
 TimeoutError: Waiting for element to be located By(css selector, [data-test-subj="mlAnnotationsActionEdit"])
 Wait timed out after 10021ms
     at /Users/trezworkbox/dev/main.worktrees/more-anomaly-detection-job-list-ui-tests/node_modules/selenium-webdriver/lib/webdriver.js:910:22
     at processTicksAndRejections (node:internal/process/task_queues:95:5)
       at onFailure (retry_for_success.ts:17:9)
       at retryForSuccess (retry_for_success.ts:84:7)
       at Proxy.try (retry.ts:51:12)
       at Proxy.clickByCssSelector (find.ts:419:5)
       at Proxy.click (test_subjects.ts:168:5)
       at MlJobTable.clickEditAnnotationAction (job_table.ts:530:7)
       at expanded_job_details.ts:54:9
       at MlJobTable.withDetailsOpen (job_table.ts:257:16)
       at Object.editAnnotation (expanded_job_details.ts:47:7)
       at Context.<anonymous> (job_list.ts:93:7)
       at Object.apply (wrap_function.js:73:16)
       at Object.apply (wrap_function.js:73:16)
```

But, it passes how I've coded it already.  Your thoughts boss man?

Reply:
Oh, my bad. I was confusing it with the job row action menu, while it's actually a nested table for annotations with separate row action menus.
So we should
* move this method to the expanded row service
* and probably also allow to address a specific row in the annotations table (there's the `mlAnnotationsTableRow ${id}` test subj we could use)
* check that the click actually opened the actions menu

Reply:
Well, it's already on a specific row boss.  Care to explain?

Reply:
https://github.com/wayneseymour/kibana/commit/d08f9778adab7688c7218aa300d85a330455eeb1